pkg/validator/validators.go

```diff
@@ -30,17 +36,49 @@ func init() {
 		Group:   "config.openservicemesh.io",
 		Version: "v1alpha1",
 	}
-	multiClusterServiceGvk := metav1.GroupVersionKind{
-		Kind:    "MultiClusterService",
-		Group:   "config.openservicemesh.io",
-		Version: "v1alpha1",
-	}
+
+	RegisterValidator(ingressBackendGvk.String(), IngressBackendValidator)
 	RegisterValidator(egressGvk.String(), EgressValidator)
 	RegisterValidator(meshConfigGvk.String(), MeshConfigValidator)
-	RegisterValidator(multiClusterServiceGvk.String(), MeshConfigValidator)
 }
 
-// EgressValidator validates the Egress CRD.
+// IngressBackendValidator validates the IngressBackend custom resource
+func IngressBackendValidator(req *admissionv1.AdmissionRequest) (*admissionv1.AdmissionResponse, error) {
+	ingressBackend := &pv1alpha1.IngressBackend{}
+	if err := json.NewDecoder(bytes.NewBuffer(req.Object.Raw)).Decode(ingressBackend); err != nil {
+		return nil, err
+	}
+
+	for _, backend := range ingressBackend.Spec.Backends {
+		// Validate port
+		switch strings.ToLower(backend.Port.Protocol) {
+		case constants.ProtocolHTTP:
+			// Valid
+
+		case constants.ProtocolHTTPS:
+			// Valid
+			// If mTLS is enabled, verify there is an AuthenticatedPrincipal specified
+			authenticatedSourceFound := false
+			for _, source := range ingressBackend.Spec.Sources {
+				if source.Kind == pv1alpha1.KindAuthenticatedPrincipal {
+					authenticatedSourceFound = true
+					break
+				}
+			}
+
+			if backend.TLS.SkipClientCertValidation && !authenticatedSourceFound {
+				return nil, errors.Errorf("HTTPS ingress with client certificate validation enabled must specify at least one 'AuthenticatedPrincipal` source")
+			}
+
+		default:
+			return nil, errors.Errorf("Expected 'Port.Protocol' to be 'http|https', got: %s", backend.Port.Protocol)
```

Comment:
Might be worth adding the full yaml/json path in the error so users also have the `spec.backends[i]` context and know right where to look.

And not a big deal, but IIRC validation errors usually use the yaml/json field names (`port.protocol`) vs. the Go struct field names (`Port.Protocol`).

I might also change `'http|https'` to `'http' or 'https'` just to make that clear. Based on the message as-is, I wouldn't fault a user for copy-pasting `http|https` into the YAML.